src/Commands/Category/CategoryProductsCount.php

```diff
@@ -0,0 +1,178 @@
+
+            $tree->printTree($output);
+        } else {
+            $fp = fopen($outputPathname, 'w');
+
+            $maxDepth = $this->findMaxDepth($rootCategory);
+            fputcsv($fp, array_merge(['ID', 'Name'], array_fill(0, $maxDepth - 1, ' '), ['Products']));
+
+            $categoriesCSV = $this->formatCategoriesToCSV($rootCategory, $maxDepth);
+            foreach ($categoriesCSV as $fields) {
+                fputcsv($fp, $fields);
+            }
+
+            fclose($fp);
+        }
+
+        return 1;
+    }
+
+    /**
+     * @param array<string, mixed> $category
+     *
+     * @return array<string|array>
+     */
+    private function formatCategoriesToTree(array $category)
+    {
+        $languageId = $this->legacyContext->getLanguage()->id;
+        $categoryLabel = $this->getCategoryLabel((int) $category['id_category'], $languageId);
+
+        if (empty($category['children'])) {
+            return [$categoryLabel => []];
+        }
+
+        $categoriesTree[$categoryLabel] = [];
+        foreach ($category['children'] as $child) {
+            $categoriesTree[$categoryLabel] = array_merge(
+                $categoriesTree[$categoryLabel],
+                $this->formatCategoriesToTree($child)
+            );
+        }
+
+        return $categoriesTree;
+    }
+
+    /**
+     * @param array<string, mixed> $category
+     *
+     * @return array<string[]>
+     */
+    private function formatCategoriesToCSV(array $category, int $maxDepth)
+    {
+        $languageId = $this->legacyContext->getLanguage()->id;
+        $categoryId = (int) $category['id_category'];
+        $depth = (int) $category['level_depth'];
+        $categoryLabel = $this->getCategoryLabel($categoryId, $languageId);
+
+        $categoriesCSV = [array_merge(
+            [$categoryId],
+            array_fill(0, $depth - 1, ' '),
+            [$categoryLabel],
+            array_fill(0, max(0, $maxDepth - $depth), ' '),
+            [(new Category($categoryId))->getProducts($languageId, 0, 0, null, null, true)]
+        )];
+
+        if (key_exists('children', $category)) {
+            foreach ($category['children'] as $child) {
+                $categoriesCSV = array_merge(
+                    $categoriesCSV,
+                    $this->formatCategoriesToCSV($child, $maxDepth)
+                );
+            }
+        }
+
+        return $categoriesCSV;
+    }
+
+    /**
+     * @param array<string, mixed> $category
+     */
+    public function findMaxDepth(array $category): int
+    {
+        if (key_exists('children', $category)) {
+            return max(array_map([$this, 'findMaxDepth'], $category['children']));
+        }
+
+        return (int) $category['level_depth'];
+    }
+
+    public function getCategoryLabel(int $categoryId, int $languageId): string
+    {
+        $category = new Category($categoryId);
+        $categoryName = $category->name[$languageId];
```

Comment:
```suggestion
        $categoryName = $category->getName($languageId);
```

or then, initialize the category with id language and shop, and you can access the name without specifying the language.